### PR TITLE
Add a basic CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by dcjones**
Adds a CONTRIBUTING.md file with a basic style guide, etc. See #46.

### CODE REVIEW

**Comment by brk00** [comment]
Nice =)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    be included from seperate files. E.g.
+
+    ```julia
+    module AwesomeFeatures
+        using Compat, JSON
+
+        include("feature1.jl")
+        include("feature2.jl")
+    end
+    ```
```
Comment (bicycle1885) [inline CONTRIBUTING.md]: Is indentation recommended within a `module` block?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  * Separate logical blocks of code with one blank line, or two blank lines for
+    function/type definitions.
+
+  * When extending method definitions, explicitly import the method.
+
+    ```julia
+    import Base: start, next done
```
Comment (bicycle1885) [inline CONTRIBUTING.md]: a comma is missing after the `next`

File: CONTRIBUTING.md (hunk 1 above)
Comment (brk00) [inline CONTRIBUTING.md]: That's not used throughout our code, and not recommended by Julia documentation.

Comment (bicycle1885) [inline CONTRIBUTING.md]: Yes. So I thought this may be a misleading example.

Comment (dcjones) [inline CONTRIBUTING.md]: My mistake. It's fixed now.

**Comment by bicycle1885** [comment]
What about `return` of function definition?

The current code seems to follow the rule:
* Do not omit `return` keywords
* except one-line definition.

**Comment by dcjones** [comment]
I think that's open to debate. I personally like explicit returns and do it out of habit now. It's only one extra word to type, and it makes it immediately obvious what a function is returning.

**Comment by Ward9250** [comment]
I agree I like explicit returns. If it's not too much effort I always prefer code to be explicit.

**Comment by bicycle1885** [comment]
I like omitting final returns but that's just my taste.
I'm not against explicit return convention.

**Comment by Ward9250** [comment]
Should we also say - we adhere to the Julia community guidelines (http://julialang.org/community/standards/) in addition to any we think of?

**Comment by Blahah** [comment]
@Ward9250 absolutely, good idea.

**Comment by dcjones** [comment]
Merged. Feel free to make PRs to amend.

**Pull request merged**